core/repos/repomanager.go

```diff
@@ -148,11 +159,13 @@ func (manager *RepoManager)  discoverRepos() (repos []string, err error){
 			}
 
 			for _, repo := range someRepos { //sanitize path
+				// Get rid of invalid paths
+				if !isValidPath(repo) { continue }
 				repoDir := manager.rService.GetReposPath()
-				utils.EnsureTrailingSlash(&repoDir)
-				repo = strings.TrimPrefix(repo, repoDir)
-				utils.EnsureTrailingSlash(&repo)
-				repos = append(repos, repo)
+				repo, err = filepath.Rel(repoDir, repo) // trim repoDir
```

Comment:
I wonder if you still need to `EnsureTrailingSlash` after this. I'm not sure, but please double check that the behavior doesn't change.

I see you remove the same call in other spots too. In principle that's ok as long as we ensure it's all internally consistent.